src/mixer/samplerbank.cpp

```diff
@@ -13,54 +13,70 @@ SamplerBank::SamplerBank(PlayerManager* pPlayerManager)
         : QObject(pPlayerManager),
           m_pPlayerManager(pPlayerManager) {
     DEBUG_ASSERT(m_pPlayerManager);
-    m_pLoadControl = new ControlPushButton(ConfigKey("[Sampler]", "LoadSamplerBank"));
-    connect(m_pLoadControl, SIGNAL(valueChanged(double)),
+    m_pCOLoadBank = new ControlPushButton(ConfigKey("[Sampler]", "LoadSamplerBank"));
+    connect(m_pCOLoadBank, SIGNAL(valueChanged(double)),
             this, SLOT(slotLoadSamplerBank(double)));
-    m_pSaveControl = new ControlPushButton(ConfigKey("[Sampler]", "SaveSamplerBank"));
-    connect(m_pSaveControl, SIGNAL(valueChanged(double)),
+    m_pCOSaveBank = new ControlPushButton(ConfigKey("[Sampler]", "SaveSamplerBank"));
+    connect(m_pCOSaveBank, SIGNAL(valueChanged(double)),
             this, SLOT(slotSaveSamplerBank(double)));
+
+    m_pCONumSamplers = new ControlProxy(ConfigKey("[Master]", "num_samplers"));
 }
 
 SamplerBank::~SamplerBank() {
-    delete m_pLoadControl;
-    delete m_pSaveControl;
+    delete m_pCOLoadBank;
+    delete m_pCOSaveBank;
 }
 
 void SamplerBank::slotSaveSamplerBank(double v) {
-    if (v == 0.0 || m_pPlayerManager == NULL) {
+    if (v <= 0.0) {
         return;
     }
-    QString filefilter = tr("Mixxx Sampler Banks (*.xml)");
+
+    QString fileFilter = tr("Mixxx Sampler Banks (*.xml)");
     QString samplerBankPath = QFileDialog::getSaveFileName(
             NULL, tr("Save Sampler Bank"),
             QString(),
             tr("Mixxx Sampler Banks (*.xml)"),
-            &filefilter);
+            &fileFilter);
     if (samplerBankPath.isNull() || samplerBankPath.isEmpty()) {
         return;
     }
+
     // Manually add extension due to bug in QFileDialog
     // via https://bugreports.qt-project.org/browse/QTBUG-27186
     // Can be removed after switch to Qt5
     QFileInfo fileName(samplerBankPath);
     if (fileName.suffix().isEmpty()) {
-        QString ext = filefilter.section(".",1,1);
+        QString ext = fileFilter.section(".",1,1);
         ext.chop(1);
         samplerBankPath.append(".").append(ext);
     }
 
+    if (!saveSamplerBankToPath(samplerBankPath)) {
+        QMessageBox::warning(NULL,
+                        tr("Error Saving Sampler Bank"),
+                        tr("Could not write the sampler bank to '%1'.")
+                        .arg(samplerBankPath));
+    }
+}
+
+bool SamplerBank::saveSamplerBankToPath(const QString& samplerBankPath) {
     // The user has picked a new directory via a file dialog. This means the
     // system sandboxer (if we are sandboxed) has granted us permission to this
     // folder. We don't need access to this file on a regular basis so we do not
     // register a security bookmark.
 
+    if (m_pPlayerManager == nullptr) {
+        qWarning() << "SamplerBank::saveSamplerBankToPath called with no PlayerManager";
```

Comment:
Thank you for documenting that. FYI, DocuWiki converts `--` into an em dash. To avoid that, it needs to be wrapped in markers to tell DocuWiki not to format it: `%%--%%`